L1Trigger/L1THGCal/plugins/be_algorithms/SingleCellClusterAlgo.cc

```diff
@@ -50,10 +63,23 @@ void SingleCellClusterAlgo::run(const l1t::HGCFETriggerDigiCollection& coll)
         {
             if(triggercell.hwPt()>0)
             {
-                cluster_product_->push_back(0,triggercell);
+
+                HGCalDetId detid(triggercell.detId());
+                int subdet =  (ForwardSubdetector)detid.subdetId();
```

Comment:
You are casting int -> ForwardSubdetector -> int. You can directly take the subdetId as an int.